tools/gluten-te/ubuntu/dockerfile-buildenv

```diff
@@ -50,14 +50,14 @@ RUN cat /root/.m2/settings.xml
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y sudo locales wget tar tzdata git ccache cmake ninja-build build-essential llvm-11-dev clang-11 libiberty-dev libdwarf-dev libre2-dev libz-dev libssl-dev libboost-all-dev libcurl4-openssl-dev
 
 # install HBM dependencies
-RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y autoconf automake g++ libnuma-dev libtool numactl unzip libdaxctl-dev
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y autoconf automake g++ libnuma-dev libtool numactl unzip libdaxctl-dev
```

Comment:
I remember that we added `apt-get update` to fix some issues that caused by out-of-date apt cache. Any reason to remove it again?

Reply:
in my environment apt update take a while so I tried to reduce duplicated invocations. I thought this is a duplication of apt update in line 50 ?

Reply:
It's a little bit related to docker build cache: the builder always caches a single `RUN ...` line as one cache layer. So if for some reason we changed line 53, a new build may still hit line 50's cache, therefore no `apt-get update` will be performed during this build.

> in my environment apt update take a while so I tried to reduce duplicated invocations

I assume it only happens at the first time to build `dockerfile-buildenv` ? Since the produced images should be cached for later builds?

Reply:
got it , thanks for the explanation. I'll revert this part